src/lib.rs

```diff
@@ -1,27 +1,51 @@
 mod generated;
 
+use async_trait::async_trait;
+
 pub use generated::info;
 pub use generated::mocap;
+pub use generated::telemetry;
+
+#[derive(Clone, Debug)]
+pub enum RequestError<PluginMavErr> {
+    MavErr(PluginMavErr),
+    RpcErr(tonic::Status),
+}
+
+pub type RequestResult<SuccessType, PluginMavErr> = std::result::Result<SuccessType, RequestError<PluginMavErr>>;
 
-use grpcio::{ChannelBuilder, EnvBuilder};
+type TonicResult<T> = std::result::Result<tonic::Response<T>, tonic::Status>;
 
+trait FromRpcResponse<T> {
+    fn from_rpc_response(rpc_result: TonicResult<T>) -> Self;
+}
 pub struct System {
     pub mocap: mocap::Mocap,
     pub info: info::Info,
+    pub telemetry: telemetry::Telemetry,
 }
 
 impl System {
-    pub fn new(url: Option<String>) -> System {
-        let env = std::sync::Arc::new(EnvBuilder::new().build());
+    pub async fn connect(url: Option<String>) -> Result<System, tonic::transport::Error> {
         let url = match url {
             Some(x) => x,
-            None => String::from("localhost:50051"),
+            None => String::from("http://0.0.0.0:50051"),
         };
-        let ch = ChannelBuilder::new(env).connect(url.as_str());
 
-        System {
-            mocap: mocap::Mocap::new(ch.clone()),
-            info: info::Info::new(ch),
-        }
+        Ok(System {
+            mocap: mocap::Mocap::connect(&url).await?,
+            info: info::Info::connect(&url).await?,
+            telemetry: telemetry::Telemetry::connect(&url).await?,
+        })
     }
 }
+
+#[async_trait]
+trait Connect {
+    async fn connect(url: &String) -> Result<Self, tonic::transport::Error> where Self: Sized;
+}
+
+#[async_trait]
+pub trait MavsdkStream<StreamRequestResult> {
+    async fn get_next(&mut self) -> Option<StreamRequestResult>;
+}
```

Comment:
This is not the best solution, as I think. Need to change it to more convenient way in future.